{
  try {
    doWork();
  }
 catch (  RemoteException e) {
    setException(e);
  }
}
